docs/api/field_types_reference/addressfield.md

```diff
@@ -0,0 +1,131 @@
+# Address Field Type
+
+This Field represents and handles address fields.
+It allows you to customize address Field Types per country.
+
+| Name      | Internal name   | Expected input              |
+|-----------|-----------------|-----------------------------|
+| `Address` | `ibexa_address` | `string`, `string`, `array` |
+
+The Address Field Type is available via the Address Bundle
+provided by the [ibexa/fieldtype-address](https://github.com/ibexa/fieldtype-address) package.
+
+## PHP API Field Type
+
+### Inputs:
+
+| Type     | Description                                   | Example           |
+|----------|-----------------------------------------------|-------------------|
+| `string` | Name of the address.                          | `My home address` |
+| `string` | Country code in ISO 3166-1 alpha-2 format.    | `PL`              |
+| `array`  | Additional fields, defined by address format. | see below         |
+
+### Example input:
+
+```php
+new FieldType\Value(
+    'My home address',
+    'PL',
+    [
+        'city' => 'Warsaw',
+        'region' => 'Masovian',
+        'postal_code' => '11-123',
+    ]
+);
+```
+
+### Validation
+
+This Field Type validates whether `Country` and `Name` fields have been field out.
+
+### Value Object:
+
+#### Properties:
+
+| Property   | Type     | Description                                   |
+|------------|----------|-----------------------------------------------|
+| `$name`    | `string` | Name of the address.                          |
+| `$country` | `string` | Country code in ISO 3166-1 alpha-2 format.    |
+| `$fields`  | `array`  | Additional fields, defined by address format. |
+
+#### Constructor:
+
+See above (Example input).
+
+### Formats:
+
+Default configuration.
+
+```yaml
+formats:
+    personal:
+        country:
+            default:
+                - region
+                - locality
+                - street
+                - postal_code
+```
+
+Defines default fields for `personal` address type.
+
+#### Extending:
+
+```yaml
+formats:
+    billing_address:
+        country:
+            DE:
+                - tax_number
+                - city
+                - address
+                - postal_code
+```
+
+Adds (or alters) an address format for `DE` country of `billing_address` type.
+
+### Field form types:
+
+By default, each field is a simple text input with a label made of field identifier.
+To change the type of field, you need to listen to a specific event.
+For each field below events are emitted (in order):
```

Comment:
```suggestion
For each field below events are dispatched (in order):
```
emit is fine, but we use dispatch elsewhere (e.g. https://doc.ibexa.co/en/master/guide/repository/event_reference/event_reference/)